lib/streamlit/widgets.py

```diff
@@ -104,26 +110,57 @@ def call_callbacks(self, new_widget_states: WidgetStates) -> None:
                 # The widget does not have an on_change callback.
                 continue
 
-            old_value = self.get_widget_value(new_state.id)
-            new_value = _get_widget_value(new_state)
-            deserializer = self._widget_deserializers.get(new_state.id, None)
-
-            if deserializer is not None:
-                old_value = deserializer(old_value)
-                new_value = deserializer(new_value)
-
-            if new_value == old_value:
-                # The widget's value has not changed.
+            if self.equivalent_values(new_state):
                 continue
 
             # The widget's value has changed - call its on_change callback.
             callback(new_value)
```

Comment:
`new_value` is not defined now.

Reply:
Trying this out in PR Preview https://share.streamlit.io/streamlit/core-previews/pr-2884

```
import streamlit as st
state = st.beta_session_state(count=0)

if st.signal('Increment'):
    increment_value = st_event.context()
    state.count += increment_value
  

selected_value = st.selectbox('Select me', ['cat', 'dog'])
st.button('Button', signal='Increment', context=1)

st.write('Counter Value = ', state.count)
```

Reply:
Looks like my editor broke, should have highlighted that `new_value` was no longer defined